src/Workspaces/Core/Portable/TemporaryStorage/TemporaryStorageServiceFactory.cs

```diff
@@ -487,7 +493,36 @@ public override int Read(char[] buffer, int index, int count)
 
                 return count;
             }
+
+            public override int ReadBlock(char[] buffer, int index, int count) =>
+                Read(buffer, index, count);
+
+#if NETCOREAPP
+            public override int Read(Span<char> buffer)
+            {
+                var count = Math.Min(buffer.Length, (int)(_end - _position));
+                if (count > 0)
+                {
+                    new ReadOnlySpan<char>(_position, count).CopyTo(buffer);
+                    _position += count;
+                }
+
+                return count;
+            }
+
+            public override int ReadBlock(Span<char> buffer) =>
+                Read(buffer);
+#endif
```

Comment:
I have overriden the `ReadBlock` methods as well because the base implementation would use a loop, which isn't needed here. I also provided the span overloads - even though they're not used in `SourceText.From` today, if they were used someday and weren't overriden here, the base implementation in `TextReader` would needlessly allocate an array and forward to the array overloads - it's a fallback back-compat implementation that is really inefficient. In my opinion, the span overloads should always be overriden to avoid that pit of failure, even if we know that the API we're passing the reader into doesn't use them right now.

In general, when implementing something like a `TextReader`, it's always a good idea to implement as many methods as possible and reasonable because we can make it more efficient than the base implementations that have to go through the other ones.